master/Master.hs

```diff
@@ -52,24 +52,32 @@ instance Show (MasterState) where
   show (MasterState skt _ cfg voteMap ackMap timeoutMap) = 
     show skt ++ show cfg ++ show voteMap ++ show ackMap ++ show timeoutMap
 
-
-
+-- Entry Point
 main :: IO ()
-main = do
-  success <- Lib.parseArguments
-  case success of
-    Nothing     -> Lib.printUsage --an error occured
-    Just config -> runKVMaster config
-
-runKVMaster :: Lib.Config -> IO ()
-runKVMaster cfg = do 
-  s <- listenOn (Lib.masterPortId cfg)
-  c <- newChan
-  mvar <- newMVar (MasterState s c cfg Map.empty Map.empty Map.empty)
-  forkIO $ runReaderT processMessages mvar
-  forkIO $ runReaderT timeoutThread mvar
+main = Lib.parseArguments >>= runKVMaster
+
+-- Initialization for Master Node
+runKVMaster :: Maybe Lib.Config -> IO ()
+runKVMaster Nothing = Lib.printUsage -- an error occured 
+runKVMaster (Just c) = do
+  s <- MasterState <$> listenOn (Lib.masterPortId c) <*> newChan
+  mvar <- newMVar (s c Map.empty Map.empty Map.empty)
+  _ <- forkIO $ runReaderT listen mvar
```

Comment:
can't you just say `forkIO $ runReaderT listen mvar` without binding the result?